### PR TITLE
[Translator] Support for AAD

### LINKED ISSUE #21306: Implement AAD

### PR DESCRIPTION
**Pull request by maririos**
The service has added AAD support :)
- Add methods to Client
- Add tests
- Updates Readme and Samples.md files

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/21306

### CODE REVIEW

**Comment by maririos** [comment]
@chidozieononiwu could you help me understand if there is something I am not setting correctly for my live tests to consistently pass when using AAD credentials? The tests are failing in different configurations and different places, and when re-running them they pass... :|

**Comment by maririos** [comment]
@chidozieononiwu I need help! all my live runs are failing in the PowerShell step. I have tried multiple times and nothing. 
Do you know what could it be?
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=923160&view=logs&jobId=6bc677fc-eaaa-5aa2-d61c-5bfd25cc27a2&j=6bc677fc-eaaa-5aa2-d61c-5bfd25cc27a2&t=78d7b733-8643-5dcd-09a9-1049ca535820

File: sdk/translation/Azure.AI.Translation.Document/samples/Sample1_StartTranslation.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ This sample demonstrates how to translate one or more documents in a blob contai
 
 ## Creating a `DocumentTranslationClient`
 
-To create a new `DocumentTranslationClient` to run a translation operation for documents, you need a Translator endpoint and credentials. In the sample below, you'll use a Translator API key credential by creating an `AzureKeyCredential` object, that if needed, will allow you to update the API key without creating a new client.
+To create a new `DocumentTranslationClient` to run a translation operation for documents, you need a Translator endpoint and credentials. You can use the [DefaultAzureCredential][DefaultAzureCredential] to try a number of common authentication methods optimized for both running as a service and development. In the sample below, you'll use a Translator API key credential by creating an `AzureKeyCredential` object, that if needed, will allow you to update the API key without creating a new client.
```
Comment (kristapratico) [inline sdk/translation/Azure.AI.Translation.Document/samples/Sample1_StartTranslation.md:6]: Doesn't seem like there's a link for [DefaultAzureCredential]

File: sdk/translation/Azure.AI.Translation.Document/README.md [hunk 2]
```diff
@@ -81,6 +81,27 @@ string apiKey = "<apiKey>";
 var client = new DocumentTranslationClient(new Uri(endpoint), new AzureKeyCredential(apiKey));
 ```
 
+#### Create DocumentTranslationClient with Azure Active Directory Credential
+
+Client API key authentication is used in most of the examples in this getting started guide, but you can also authenticate with Azure Active Directory using the [Azure Identity library][azure_identity].  Note that regional endpoints do not support AAD authentication.
+
+Create a [custom subdomain][custom_subdomain] for your resource in order to use this type of authentication.  
+
+To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, or other credential providers provided with the Azure SDK, please install the Azure.Identity package:
```
Comment (kristapratico) [inline sdk/translation/Azure.AI.Translation.Document/README.md:90]: check the links here, doesn't seem like they exist

**Review by kristapratico — APPROVED** [review]

File: sdk/translation/Azure.AI.Translation.Document/samples/Sample1_StartTranslation.md (hunk 1 above)
Comment (maririos) [reply sdk/translation/Azure.AI.Translation.Document/samples/Sample1_StartTranslation.md:6]: plop u are right. Thank you!
I added them to all samples.md files

File: sdk/translation/Azure.AI.Translation.Document/README.md (hunk 2 above)
Comment (maririos) [reply sdk/translation/Azure.AI.Translation.Document/README.md:90]: double plop.
Thank you! added

**Pull request merged**